### PR TITLE
Check if a default kubeconfig file path exists

### PR DESCRIPTION
**Pull request by qianz2**
**Task name**: HelmDeployV0

**Description**: Check if a default file path $HOME/.kube/config file exists before throwing errors.

**Documentation changes required:** (Y/N) <Please mark if documentation changes are required>

**Added unit tests:** (Y/N) <Please mark if unit tests were added or updated according changes>

**Attached related issue:** (Y/N) <Please add link to related issue here>

**Checklist**:
- [ ] Task version was bumped - please check [instruction](https://github.com/microsoft/azure-pipelines-tasks/tree/master/docs/taskversionbumping.md) how to do it
- [ ] Checked that applied changes work as expected

### CODE REVIEW

**Comment by merlynomsft** [comment]
Tagging the bug #18666

File: _generated/HelmDeployV0/src/helm.ts [hunk 1]
```diff
@@ -84,7 +84,11 @@ async function run() {
     var kubectlCli: kubernetescli;
     var externalAuth = connectionType === "None" && (command === "install" || command === "upgrade");
     if (externalAuth && !tl.getVariable("KUBECONFIG")) {
-        tl.error("KUBECONFIG kube configuration file path must be set when connectionType is none and command is install or upgrade.");
+        if (fs.existsSync("$HOME/.kube/config")) {
```
Comment (merlynomsft) [inline _generated/HelmDeployV0/src/helm.ts:87]: I'm not sure about the $HOME expansion here.  (just double checking that it was tested+works)

File: _generated/HelmDeployV0/src/helm.ts [hunk 2]
```diff
@@ -84,7 +84,11 @@ async function run() {
...
+            tl.setVariable("KUBECONFIG", "$HOME/.kube/config");
```
Comment (merlynomsft) [inline _generated/HelmDeployV0/src/helm.ts:88]: Not blocking, but I still have a question about how this variable reportedly worked in 0.223.1 without KUBECONFIG set

**Review by merlynomsft — APPROVED** [review]

File: _generated/HelmDeployV0/src/helm.ts (hunk 1 above)
Comment (tkasparek) [reply _generated/HelmDeployV0/src/helm.ts:87]:
```
[root@DESKTOP-OA5AB39 ~]# ls -l $HOME/test.sh
-rwxr-xr-x 1 root root 156 Mar  3 16:00 /root/test.sh
[root@DESKTOP-OA5AB39 ~]# node
Welcome to Node.js v20.4.0.
Type ".help" for more information.
> fs = require('fs');
> fs.existsSync("$HOME/test.sh")
false
> fs.existsSync("/root/test.sh")
true
```

The variable expansion does not work.

**Review by tkasparek — CHANGES_REQUESTED** [review]

File: _generated/HelmDeployV0/src/helm.ts (hunk 1 above)
Comment (qianz2) [reply _generated/HelmDeployV0/src/helm.ts]: Updated and tested:

> fs = require('fs');
> path = require('path');
> const kubeConfigPath = path.join(process.env.HOME, '.kube', 'config');
undefined
> fs.existsSync(kubeConfigPath)
true

Comment (qianz2) [reply _generated/HelmDeployV0/src/helm.ts]: @tkasparek  Could you take a look at it again? Thanks

**Review by tkasparek — APPROVED** [review]

**Pull request merged**